packages/compiler/formatter/print/printer.ts

```diff
@@ -810,10 +811,13 @@ function isStringSafeToUnquote(id: StringLiteralNode, options: CadlPrettierOptio
 }
 
 function isModelExpressionInBlock(path: AstPath<ModelExpressionNode>) {
-  const parent: Node | null = path.getParentNode() as any;
+  // The parent can either be a regular Node or an OperationSignature because
+  // the printer has to walk through `OperationSignatureNode.signature` which
+  // isn't a node itself.
+  const parent: Node | OperationSignature | null = path.getParentNode() as any;
 
   switch (parent?.kind) {
-    case SyntaxKind.OperationStatement:
+    case "OperationDeclaration":
```

Comment:
Why isn't this in SyntaxKind enum?

Reply:
The `OperationSignature` type isn't a node type, it's a union for another set of auxiliary types `OperationSignatureDeclaration` and `OperationSignatureReference`.  The `kind` field of those is just a string of `OperaitonDeclaration` or `OperationReference`.  

I had to tweak this logic a bit due to how the formatter APIs work, because including `signatures` in the path caused it to look at the `OperationSignature` type, this requiring a different type for `kind` to be considered in the switch.

Reply:
I'll try to convert it to a node pattern to see if it works, but I think it might complicate the code in other ways.  

Previous to this design, I actually did add a separate syntax node for the "operation reference" style and Brian wasn't thrilled by it.  It's slightly different than what you're suggesting though, so maybe it's kosher.  Definitely need to get Brian's feedback on that.

Reply:
Check this commit for an alternate implementation using nodes for the signature types:

https://github.com/daviwil/cadl/commit/b7e5f674f31e37feb9d4bd18376236211b81e449

@bterlson, do you prefer this approach or the one I'm already using in the PR?

Reply:
yeah I think we should stick to node types here or it should just be an inline property of another node.

Reply:
OK, I pushed the changes from that branch into the PR, everything is converted over to nodes now.